src/daemon/daemon.cpp

```diff
@@ -2012,15 +2056,14 @@ void mp::Daemon::create_vm(const CreateRequest* request, grpc::ServerWriter<Crea
 {
     auto checked_args = validate_create_arguments(request, *config->factory);
 
-    auto timeout = request->timeout() > 0 ? std::chrono::seconds(request->timeout()) : mp::default_timeout;
-
     if (!checked_args.option_errors.error_codes().empty())
     {
         return status_promise->set_value(grpc::Status(grpc::StatusCode::INVALID_ARGUMENT, "Invalid arguments supplied",
                                                       checked_args.option_errors.SerializeAsString()));
     }
 
-    auto name = name_from(checked_args.instance_name, *config->name_generator, vm_instances);
+    auto name = name_from(checked_args.instance_name, config->workflow_provider->name_from_workflow(request->image()),
+                          *config->name_generator, vm_instances);
```

Comment:
Yes, I went round and round about this and `timeout`, but that will take a non-trivial amount of refactoring.  Instead, it was much more straightforward to add some accessors for these bits of data.  I will add a TODO at this point since you are OK with that :wink:

Reply:
Added `TODO`'s here and for timeout.